internal/wasm/store.go

```diff
@@ -477,30 +478,44 @@ func (s *Store) Instantiate(module *Module, name string) (*PublicModule, error)
 			return nil, fmt.Errorf("module[%s] start function failed: %w", name, err)
 		}
 	}
-	return &PublicModule{s, instance}, nil
+	return &PublicModule{s: s, instance: instance}, nil
 }
 
-func (s *Store) ReleaseModuleInstance(instance *ModuleInstance) error {
+// ReleaseModuleInstance deallocates resources if a module with the given name exists.
+func (s *Store) ReleaseModuleInstance(moduleName string) error {
```

Comment:
using a string name here is fine because module name uniqueness is guaranteed regardless of host or wasm module. later, this will allow us to call easier from an outside package (perhaps needing to export the name field in PublicXX)